src/wallet/wallet.cpp

```diff
@@ -2515,6 +2515,12 @@ bool CWallet::SelectCoins(const std::vector<COutput>& vAvailableCoins, const CAm
 
     // form groups from remaining coins; note that preset coins will not
     // automatically have their associated (same address) coins included
+    if (coin_control.m_avoid_partial_spends && vCoins.size() > 10) {
```

Comment:
can we make this 10 a constant to make it easier for the reader/future writer, or make it a parameter to the grouping function itself?

Reply:
Second thought: Would an unconditional shuffle just be easier here? We shuffle inputs later post-selection, so nothing of value should be lost.

Reply:
I haven't checked, but I don't think you can remove the later shuffle, and you at least want to skip shuffling if `!m_avoid_partial_spends`, so I don't think unconditional is ideal here.

I.e. if we could *move* the shuffle operation to always happen here, rather than later, that would work fine.

If we can't, we should at least not skip the extra shuffle if we are not grouping, since that's just no-reason-wasteful.